RND-81989 > Add possibility to check compilation errors for install-application command

maksymkopachevskyi opened the pull request.

maksymkopachevskyi committed:
RND-81989 forceInstall for Install app command from Clio

maksymkopachevskyi committed:
RND-81989 checkCompilationErrors option for Install app command from Clio

maksymkopachevskyi committed:
Hide check-compilation-errors option

vladimir-nikonov commented on clio/Package/ApplicationInstaller.cs:
```diff
@@ -51,16 +56,19 @@ private bool InternalUnInstall(SysInstalledApp appInfo, EnvironmentSettings envi
 
 		protected override string GetRequestData(string fileName, PackageInstallOptions packageInstallOptions){
 			string code = _fileSystem.GetFileNameWithoutExtension(new FileInfo(fileName));
-			return
```
Keep the backward compatibility with the oldest Creatio version, in case the check for compilation errors is not used

vladimir-nikonov committed:
feat: Add CreateDevEnvironmentCommand for macOS K8s dev environment setup

- Implemented CreateDevEnvironmentCommand with 7-stage pipeline (validate, infrastructure, deploy, configure, database, dev-mode, finalize)
- Added 9 configurable options: zip, target, env-name, maintainer, port, username, password, skip-infra, no-confirm
- Created IKubernetesService, IConfigPatcherService, IPostgresService interfaces for service-oriented design
- Registered command and services in dependency injection container (BindingsModule)
- Added command to Program.cs for CLI routing and discovery
- Interactive prompt for target directory with current directory as default
- Proper error handling and validation with user confirmation flow

vladimir-nikonov committed:
Revert "feat: Add CreateDevEnvironmentCommand for macOS K8s dev environment setup"

This reverts commit 492b2171641a24e1051a534bf425b7cb11374567.

vladimir-nikonov commented on clio/Package/ApplicationInstaller.cs:
```diff
@@ -51,16 +56,19 @@ private bool InternalUnInstall(SysInstalledApp appInfo, EnvironmentSettings envi
+			return _checkCompilationErrors 
+				? $"{{\"Name\":\"{code}\",\"Code\":\"{code}\",\"ZipPackageName\":\"{fileName}\",\"LastUpdateString\":0,\"CheckCompilationErrors\":true}}"
+				: $"{{\"Name\":\"{code}\",\"Code\":\"{code}\",\"ZipPackageName\":\"{fileName}\",\"LastUpdateString\":0}}";
 		}
```
transfer this to using DTO with the correct JSON options, and we need to use tree state because it will prevent providing CheckCompilation force to false

maksymkopachevskyi replied:
Fixed, thanks

maksymkopachevskyi committed:
RND-81989 Update installation process to support optional compilation error checking
